sherpa/astro/ui/utils.py

```diff
@@ -1438,28 +1435,23 @@ def load_ascii_with_errors(self, id, filename=None, colkeys=None, sep=' ',
                                             colkeys=colkeys,
                                             dstype=Data1DAsymmetricErrs,
                                             sep=sep, comment=comment))
-        datas = self.get_data(id)
-        def calc_err(data):
-            return data.y - data.elo, data.ehi - data.y
 
-        def calc_staterror(data):
-            if func is numpy.average:
-                return func([data.elo, data.ehi], axis=0)
-            else:
-                return func(data.elo, data.ehi)
+        data = self.get_data(id)
+
+        if not delta:
+            data.elo = data.y - data.elo
+            data.ehi = data.ehi - data.y
 
-        if type(datas) is list:
-            for data in datas:
-                if delta is False:
-                    data.elo, data.ehi = calc_err(data)
-                data.staterror = calc_staterror(data)
+        if func is numpy.average:
+            staterror = func([data.elo, data.ehi], axis=0)
```

Comment:
I think at this point we should be able to just access the attributes without data chackes, and rely on the setters for the data object to do the validation. Otherwise every place you use `data.elo` you need to add a check

Reply:
OK.